eng/scripts/Show-FailureLogs.ps1

```diff
@@ -15,14 +16,28 @@ if (!$filteredLogs) {
     exit 0
 }
 
-foreach ($logFile in $filteredLogs)
+$filteredLogs = $filteredLogs | foreach {$_.FullName}
+
+for ($i = 0; $i -lt $filteredLogs.Length; $i += 1)
 {
+    $logFile = $filteredLogs[$i]
+
     Write-Host "//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////"
     Write-Host "=============================================================================================================================="
     Write-Host "Log file: $logFile"
     Write-Host "=============================================================================================================================="
     try {
         Get-Content $logFile | Write-Host
+
+        if ($i -lt $vcpkgLogFileNames.Length)
+        {
+            $rawContents = Get-Content $logFile -Raw
+            $regexMatches = Select-String "See logs for more information\:\s*(\r|\n|\r\n|\n\r)(\s+(?<logFilePath>\S*)\s*(\r|\n|\r\n|\n\r))+" -input $rawContents -AllMatches
+            foreach ($additionalLogFile in $regexMatches.matches.groups.Where({ $_.Name -eq "logFilePath" }))
```

Comment:
Modifying the collection of files in the loop (`filteredLogs`), while we are iterating that collection is a bit tricky to understand.

Can you add a comment explaining what you are trying to do here in this loop, and why, for future readers.